### PR TITLE
GUVNOR-2298: Project Explorer - No longer possible to navigate to Project root when using Repository View

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/GUVNOR-2298

### CODE REVIEW

File: kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-client/src/main/java/org/kie/workbench/common/screens/explorer/client/widgets/BaseViewPresenter.java [hunk 1]
```diff
@@ -157,7 +157,11 @@ public void onBranchSelected( String branch ) {
     }
 
     public void onActiveOptionsChange( @Observes ActiveOptionsChangedEvent changedEvent ) {
-        setVisible( isViewVisible() );
+        final boolean isVisible = isViewVisible();
+        setVisible( isVisible );
+        if ( isVisible ) {
+            initialiseViewForActiveContext( context );
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-client/src/main/java/org/kie/workbench/common/screens/explorer/client/widgets/BaseViewPresenter.java]: @csadilek This is the fix for all issues! ```FolderListing``` holds either a ```Package``` (Business View) or ```Path``` (Technical View). This critical difference drives how the Project Explorer renders content and what behaviour is enabled (i.e. "up" to parent folder icon) etc. Unless we re-initialise the ```FolderListing``` object for the correct View everything breaks. The bug was introduced by this https://github.com/droolsjbpm/kie-wb-common/commit/affac7a08d9f6d9ad1f3a32950fd5428db235b95

Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-client/src/main/java/org/kie/workbench/common/screens/explorer/client/widgets/BaseViewPresenter.java]: Prior to the commit we used to call ```initialiseViewForActiveContext``` when switching Views. See here https://github.com/droolsjbpm/kie-wb-common/commit/affac7a08d9f6d9ad1f3a32950fd5428db235b95#diff-fb5cdfd301097409ed2dddeec03b10c2L136

**Comment by csadilek** [comment]
+1

**Comment by csadilek** [comment]
Thanks!

master: dbc7955cb061bb5165881e79a98c0c8065f77134

**Pull request closed without merging**